modules/ensemble/lib/action/device_security.dart

```diff
@@ -0,0 +1,94 @@
+import 'package:ensemble/framework/action.dart';
+import 'package:ensemble/framework/scope.dart';
+import 'package:ensemble/screen_controller.dart';
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+import 'package:root_jailbreak_sniffer/rjsniffer.dart';
+import 'package:ensemble_ts_interpreter/invokables/invokable.dart';
+import 'package:ensemble/framework/event.dart';
+
+class DeviceSecurity extends EnsembleAction with Invokable {
+  EnsembleAction? onSuccess;
+  EnsembleAction? onError;
+
+  DeviceSecurity({
+    this.onSuccess,
+    this.onError,
+  });
+
+  @override
+  Future<void> execute(BuildContext context, ScopeManager scopeManager) async {
+    if (kIsWeb) {
+      _handleSuccess(context, false, false, false);
+      return;
+    }
+
+    try {
+      // Check if the device is rooted, debugged, or an emulator
+      bool isRooted = await Rjsniffer.amICompromised() ?? false;
+      bool isDebugged = await Rjsniffer.amIDebugged() ?? false;
+      bool isEmulator = await Rjsniffer.amIEmulator() ?? false;
+
+      _handleSuccess(context, isRooted, isDebugged, isEmulator);
+    } catch (e) {
+      _handleError(context, e);
+    }
+  }
+
+  void _handleSuccess(
+      BuildContext context, bool isRooted, bool isDebugged, bool isEmulator) {
+    if (onSuccess != null) {
+      ScreenController().executeAction(
+        context,
+        onSuccess!,
+        event: EnsembleEvent(
+          this,
```

Comment:
add a message field. For web, message would say something like 'this information is not available on the web' so at least the user would know. For other platforms, it can just say success